provider/oci/images.go

```diff
@@ -190,27 +203,31 @@ func (i ImageCache) ImageMetadata(base corebase.Base, defaultVirtType string) []
 				val.ImageType = ImageTypeVM
 			}
 		}
-		imgMeta := &imagemetadata.ImageMetadata{
-			Id:   val.Id,
-			Arch: "amd64",
-			// TODO(gsamfira): add region name?
-			// RegionName: region,
-			Version:  val.Base.Channel.Track,
-			VirtType: string(val.ImageType),
+		// Iterate over the list of supported architectures of the
+		// image.
+		for _, arch := range val.Archs() {
```

Comment:
Why do we need to iterate over val.Arches(), if val can only be 1 arch?

Reply:
the issue is that we cannot enforce that. Since the `instanceTypes` are created from the returned list of shapes from the API, we can't be sure that it will always be the case. You suggest that we throw an error if `len(val.Archs()) > 1`?